stdlib/LinearAlgebra/src/blas.jl

```diff
@@ -964,6 +965,74 @@ Return the updated `y`.
 """
 sbmv!
 
+### spmv!, (SP) symmetric packed matrix-vector operation defined as y := alpha*A*x + beta*y.
+for (fname, elty) in ((:dspmv_, :Float64),
+                      (:sspmv_, :Float32))
+    @eval begin
+        function spmv!(uplo::AbstractChar,
+                       n::BlasInt,
+                       α::$elty,
+                       AP::Union{Ptr{$elty}, AbstractArray{$elty}},
+                       x::Union{Ptr{$elty}, AbstractArray{$elty}},
+                       incx::Integer,
+                       β::$elty,
+                       y::Union{Ptr{$elty}, AbstractArray{$elty}},
+                       incy::Integer)
+
+            ccall((@blasfunc($fname), libblas), Cvoid,
+                  (Ref{UInt8},     # uplo,
+                   Ref{BlasInt},   # n,
+                   Ref{$elty},     # α,
+                   Ptr{$elty},     # AP,
+                   Ptr{$elty},     # x,
+                   Ref{BlasInt},   # incx,
+                   Ref{$elty},     # β,
+                   Ptr{$elty},     # y, out
+                   Ref{BlasInt}),  # incy
+                  uplo,
+                  n,
+                  α,
+                  AP,
+                  x,
+                  incx,
+                  β,
+                  y,
+                  incy)
+        end
+    end
+end
+
+function spmv!(uplo::AbstractChar,
+               α::Number, AP::Union{DenseArray{T}, AbstractVector{T}}, x::Union{DenseArray{T}, AbstractVector{T}},
+               β::Number, y::Union{DenseArray{T}, AbstractVector{T}}) where {T <: BlasReal}
+    require_one_based_indexing(AP, x, y)
+    N = length(x)
+    if N != length(y)
+        throw(DimensionMismatch("x has length $(N), but y has length $(length(y))"))
+    end
+    if length(AP) < Int64(N*(N+1)/2)
+        throw(DimensionMismatch("Packed symmetric matrix A has size smaller than length(x) = $(N)."))
+    end
+    GC.@preserve x y AP spmv!(uplo, N, convert(T, α), AP, pointer(x), BlasInt(stride(x, 1)), convert(T, β), pointer(y), BlasInt(stride(y, 1)))
```

Comment:
Do we need to preserve `AP` explicitly? I guess it's referenced directly and not in danger of being gc'ed.

Reply:
Maybe I'm too cautious here due to lack of deep GC knowledge, so looking at the other @GC.preserve statements in `blas.jl` I do not see any crucial difference and reason to drop AP from the preserve statement. Could you maybe explain this a bit more or point out the the thing that I'm missing?

Reply:
IIUC, then only those are listed which are wrapped by a `pointer(...)`. You could double-check if that's the case, but that was my understanding.

Reply:
> IIUC, then only those are listed which are wrapped by a `pointer(...)`. You could double-check if that's the case, but that was my understanding.

So if I were to drop the `pointer(...)` around x and y as well then the whole @GC.preserve might be unnecessary? If true this might be tempting to apply to have a more clean API. :)

Reply:
Possibly following here a pattern similar to that in the  `symv!` implementation strengthens your reading so removing the `@GC.preserve` alltogether might good to go. If you are also okay with it I will apply this modification as well.

Reply:
On the other hand maybe the following is a non-issue but wherever we have a `@GC.preserve` we also have Union{DenseArray,...} or Union{Unitrange,...} or Union{.., ...}. Maybe these are unrelated but they do appear together so the removal of `@GC.preserve` would  induce to suggest to remove `pointer(...)` and `GC.preserve` from all those functions. I don't know whether this is wise as people might have put it this way for a reason unknown to me at the moment.

Reply:
Probably there is no harm in keeping `AP` here, so unless someone comes by and helps in resolving this aspect, this should not keep us from merging this.

Reply:
I am planning other similar pull requests. Would it be okay  in your opinion to ask someone within this comment thread  whether to use the `@GC.preserve` to ease further development?

Reply:
@dkarrasch

Reply:
It's better just to pass the arrays, see https://github.com/JuliaLang/julia/pull/34783.